server/db_component.cpp

```diff
@@ -333,36 +336,70 @@ bool DB::Initialized()
 	return connection != nullptr;
 }
 
-string DB::PathFromID(unsigned int id, string * title, string * artist)
+string DB::PathFromID(unsigned int id, string * title, string * artist, string nspace)
 {
 	string rv;
 	assert(Initialized());
-	string sql("select path,title,artist from tracks where id = " + to_string(id) + ";");
+	string sql("select parent,title,artist,fname from tracks where id = " + to_string(id) + " and namespace = \'" + nspace + "\';");
+	cerr << sql << endl;
 	sql::Statement * stmt = connection->createStatement();
 	sql::ResultSet *res = nullptr;
+	int up;
 
 	try
 	{
 		if (stmt == nullptr)
 			throw LOG(_log_, "createStatement() failed");
 
 		res = stmt->executeQuery(sql.c_str());
+
 		if (res->next())
 		{
-			rv = res->getString("path");
+			up = res->getInt("parent");
 			if (title != nullptr)
 				*title = res->getString("title");
 			if (artist != nullptr)
 				*artist = res->getString("artist");
+			rv = res->getString("fname");
 		}
+		cerr << *title << endl;
+		cerr << *artist << endl;
 	}
 	catch (LoggedException s)
 	{
+		throw s;
```

Comment:
If the `stmt == nullptr` is the only call that will cause a `LoggedException` you can remove that check from the try-catch.